tools/integration_tests/util/client/storage_client.go

```diff
@@ -222,3 +223,59 @@ func StatObject(ctx context.Context, client *storage.Client, object string) (*st
 	}
 	return attrs, nil
 }
+
+func UploadGcsObject(localPath, bucketName, objectName string, uploadGzipEncoded bool) error {
+	ctx := context.Background()
+
+	client, err := storage.NewClient(ctx)
```

Comment:
Instead of creating a new storage client here each time, pass the context and the client in the function just like in all the other utilities above.